linera-views/src/scylla_db.rs

```diff
@@ -328,41 +335,121 @@ impl ScyllaDbClientInternal {
         client.1.clone()
     }
 
+    /// Test if a table is present in the database or not
+    pub async fn test_existence_table(
+        session: &Session,
+        table_name: &String,
+    ) -> Result<bool, ScyllaDbContextError> {
+        // We check the way the test can fail. It can fail in different ways.
+        let query = format!("SELECT table_name FROM kv.pairs_{}", table_name);
+
+        // Execute the query
+        let result = session.query(query, &[]).await;
+
+        // The missing table translates into a very specific error that we matched
+        let error_miss1 =
+            QueryError::DbError(Invalid, format!("unconfigured table pairs_{}", table_name));
+        let error_miss2 = QueryError::DbError(
+            Invalid,
+            "Undefined name table_name in selection claus".to_string(),
+        );
+        let error_miss3 = QueryError::DbError(Invalid, "Keyspace kv does not exist".to_string());
+        let Err(error) = result else {
+            // If ok, then the tables
+            return Ok(true);
+        };
+        let error_str = format!("{:?}", error);
+        let error_miss1_str = format!("{:?}", error_miss1);
+        let error_miss2_str = format!("{:?}", error_miss2);
+        let error_miss3_str = format!("{:?}", error_miss3);
+        if error_str == error_miss1_str
+            || error_str == error_miss2_str
+            || error_str == error_miss3_str
+        {
+            Ok(false)
+        } else {
+            Err(ScyllaDbContextError::ScyllaDbQueryError(error))
+        }
+    }
+
+    /// Create a table, the keyspace might or might not be existing.
+    ///
+    /// For the table itself, we return true if the table already exists
+    /// and false othersize.
+    /// If the stop_if_table_exists is put then the existence of a table
+    /// triggers an error.
+    ///
+    /// This is done since the test_existence_table might return true but
+    /// be called
```

Comment:
I don't understand this sentence.

Reply:
Changed to **This is done since if new is called two times the `test_existence_table` might be called two times which would lead to create_table called two times.**